cmd/influx_tsm/main.go

```diff
@@ -36,96 +39,107 @@ Convert a database from b1 or bz1 format to tsm1 format.
+name with '.%v'. The backups will be ignored by the system since they
 are not registered with the cluster.
 
 To restore a backup, delete the tsm1 version, rename the backup directory
 restart the node.`, backupExt)
 
-var dataPath string
-var ds string
-var tsmSz uint64
-var parallel bool
-var disBack bool
-
-const maxTSMSz = 2 * 1000 * 1000 * 1000
+type options struct {
+	DataPath       string
+	DBs            []string
+	TSMSize        uint64
+	Parallel       bool
+	SkipBackup     bool
+	UpdateInterval time.Duration
+	Quiet          bool
+}
 
-func init() {
-	flag.StringVar(&ds, "dbs", "", "Comma-delimited list of databases to convert. Default is to convert all databases.")
-	flag.Uint64Var(&tsmSz, "sz", maxTSMSz, "Maximum size of individual TSM files.")
-	flag.BoolVar(&parallel, "parallel", false, "Perform parallel conversion.")
-	flag.BoolVar(&disBack, "nobackup", false, "Disable database backups. Not recommended.")
-	flag.Usage = func() {
-		fmt.Fprintf(os.Stderr, "Usage: %s [options] <data-path> \n", os.Args[0])
-		fmt.Fprintf(os.Stderr, "%s\n\n", description)
-		flag.PrintDefaults()
+func (o *options) Parse() error {
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+
+	var dbs string
+
+	fs.StringVar(&dbs, "dbs", "", "Comma-delimited list of databases to convert. Default is to convert all databases.")
+	fs.Uint64Var(&opts.TSMSize, "sz", maxTSMSz, "Maximum size of individual TSM files.")
+	fs.BoolVar(&opts.Parallel, "parallel", false, "Perform parallel conversion. (up to GOMAXPROCS shards at once)")
+	fs.BoolVar(&opts.SkipBackup, "nobackup", false, "Disable database backups. Not recommended.")
+	fs.BoolVar(&opts.Quiet, "quiet", false, "Suppresses the regular status updates.")
+	fs.DurationVar(&opts.UpdateInterval, "interval", 5*time.Second, "How often status updates are printed.")
+	fs.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage: %v [options] <data-path> \n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "%v\n\n", description)
+		fs.PrintDefaults()
 		fmt.Fprintf(os.Stderr, "\n")
 	}
-}
 
-func main() {
-	pg := NewParallelGroup(1)
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		return err
+	}
 
-	flag.Parse()
-	if len(flag.Args()) < 1 {
-		fmt.Fprintf(os.Stderr, "No data directory specified\n")
-		os.Exit(1)
+	if len(fs.Args()) < 1 {
+		return errors.New("no data directory specified")
 	}
-	dataPath = flag.Args()[0]
+	o.DataPath = fs.Args()[0]
 
-	if tsmSz > maxTSMSz {
-		fmt.Fprintf(os.Stderr, "Maximum TSM file size is %d\n", maxTSMSz)
-		os.Exit(1)
+	if o.TSMSize > maxTSMSz {
+		return fmt.Errorf("bad TSM file size, maximum TSM file size is %d", maxTSMSz)
 	}
 
-	// Check if specific directories were requested.
-	reqDs := strings.Split(ds, ",")
-	if len(reqDs) == 1 && reqDs[0] == "" {
-		reqDs = nil
+	// Check if specific databases were requested.
+	o.DBs = strings.Split(dbs, ",")
+	if len(o.DBs) == 1 && o.DBs[0] == "" {
+		o.DBs = nil
+	}
+
+	return nil
+}
+
+var opts options
+
+const maxTSMSz = 2 * 1000 * 1000 * 1000
```

Comment:
Will do, I wasn't sure if 1000 was chosen for a particular reason, so I left it.